crud.py

```diff
@@ -0,0 +1,75 @@
+from sqlalchemy.orm import Session
+import models
+import schemas
+from fastapi import HTTPException
+
+
+def get_all_authors(
+        db: Session,
+        skip: int | None = None,
+        limit: int | None = None,
+) -> list[models.Author]:
+    query = db.query(models.Author)
+    if limit:
+        query = query.limit(limit)
+    if skip:
+        query = query.offset(skip)
+    return query.all()
+
+
+def create_author(db: Session, author: schemas.AuthorCreate) -> models.Author:
+    db_author = models.Author(
+        name=author.name,
+        bio=author.bio
+    )
+    db.add(db_author)
+    db.commit()
+    db.refresh(db_author)
+
+    return db_author
+
+
+def get_author_detail(db: Session, id: int) -> models.Author:
+    author = db.query(models.Author).filter(models.Author.id == id).first()
+    if author is None:
+        raise HTTPException(
+            status_code=404,
+            detail=f"Author with id {id} not found!"
+        )
+    return author
+
+
+def get_all_books(
+        db: Session,
+        skip: int | None = None,
+        limit: int | None = None,
```

Comment:
Ensure compatibility with Python versions below 3.10 by using `Optional[int]` for type hinting instead of `int | None`.